go/core/flow.go

```diff
@@ -164,6 +171,27 @@ type retryInstruction struct {
 	FlowID string `json:"flowId,omitempty"`
 }
 
+// flowInstructioner is the common type of all flowInstruction[I] types.
+type flowInstructioner interface {
+	IsFlowInstruction()
```

Comment:
I think all these methods can be unexported.

Reply:
This is actually not needed at all anymore. I was playing around with something before but it's not necessary. Reverted.